folio-spring-i18n/src/main/java/org/folio/spring/model/TranslationMap.java

```diff
@@ -0,0 +1,185 @@
+package org.folio.spring.model;
+
+import com.ibm.icu.text.MessageFormat;
+import jakarta.annotation.Nullable;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * This object represents the relationship between a locale and a
+ * {@link TranslationFile TranslationFile}.  The quality of this relationship is determined by how
+ * well the file's locale matches up with the one in use.  This object will also contain fallbacks,
+ * such as alternative languages for the user, or the server's default locale, to provide more
+ * seamless support for missing translations.
+ */
+@Log4j2
+@ToString
+@EqualsAndHashCode
+public class TranslationMap {
+
+  @Getter
+  protected final Locale locale;
+
+  @Getter
+  protected final TranslationMatchQuality quality;
+
+  @Getter
+  protected final TranslationFile file;
+
+  protected final Map<String, String> patterns;
+
+  @Getter
+  @Nullable
+  protected final TranslationMap fallback;
```

Comment:
These fields all seem like they should be `private`

Reply:
They could be; I prefer to leave things open for future extension (especially for a reusable library like this).

Reply:
Counter-point: We can always open up the visibility later on really easily if needed. But locking it down later is a breaking change in a lib, so it's much harder

Reply:
Gotcha — I've had the opposite experience with shared libraries in the past 🙃 but not sure that's applicable here.

I'll make the TranslationMap and supporting classes private, but leave the service open